src/pages/VeSDL/OnChainVote.tsx

```diff
@@ -0,0 +1,247 @@
+        } else if (selectedGauge?.address && gaugeControllerContract) {
+          const txn = await gaugeControllerContract.vote_for_gauge_weights(
+            selectedGauge.address,
+            Math.round(parseFloat(voteWeightToSubmit) * 100), //convert percent to bps
+          )
+          await enqueuePromiseToast(chainId, txn.wait(), "vote")
+          setSelectedGauge(null) // Initialize selectedGauge
+          setVoteWeightToSubmit("") // Initialize vote weight input
+        }
+      } catch (error) {
+        console.error("error on vote ==>", error)
+      }
+    }
+  }
+
+  return (
+    <React.Fragment>
+      <Stack spacing={2} px={2} pt={2} mb={2}>
+        <Typography variant="h2" textAlign="center">
+          {t("voteForNextWeek")}
+        </Typography>
+        <Box display="flex" justifyContent="space-between">
+          <Box display="flex">
+            <Typography component="label">
+              {t("yourVotingPower")}:
+              <Typography component="span" mx="8px">
+                {commify(formatBNToString(veSdlBalance, 18, 2))}
+              </Typography>
+              veSDL
+            </Typography>
+          </Box>
+          <Link
+            sx={{
+              color: theme.palette.text.primary,
+              textDecorationColor: theme.palette.text.primary,
+              whiteSpace: "nowrap",
+            }}
+          >
+            {t("gaugeDoc")}
+          </Link>
+        </Box>
+        {alertMessage && (
+          <Alert severity="error">
+            <Typography textAlign="center">{alertMessage}</Typography>
+          </Alert>
+        )}
+        <Box
+          border={`1px solid ${theme.palette.other.divider}`}
+          px={2}
+          pt={2}
+          borderRadius="10px"
+        >
+          <Autocomplete
+            id="gauge-names"
+            options={Object.keys(gaugeNames).map((key) => gaugeNames[key])}
+            getOptionLabel={(option) => option?.gaugeName ?? ""}
+            popupIcon={<ArrowDownIcon />}
+            renderInput={(params) => (
+              <TextField
+                variant="standard"
+                {...params}
+                placeholder={t("chooseGauge")}
+              />
+            )}
+            onChange={(event, newGaugeName) => {
+              if (newGaugeName) {
+                setSelectedGauge(newGaugeName)
+              }
+            }}
+            value={selectedGauge}
+            // check whether the current value belong to options
+            isOptionEqualToValue={(option, value) =>
+              option?.address === value?.address
+            }
+          />
+          <Box display="flex" gap={4} my={2}>
+            <TextField
+              variant="standard"
+              label={t("votingWeight")}
+              type="text"
+              value={voteWeightToSubmit}
+              onChange={(event) => {
+                if (isNumberOrEmpty(event.target.value)) {
+                  const estimatedWeight =
```

Comment:
Does this work? I'd expect JS to use the standard order of operations in math: parenthesis, multiplication, division, addition, subtraction. eg

`99 + 99 / 100 = 99.99`